### PR TITLE
fix(recovery): stop routine disposition recovery from starting work

### PR DESCRIPTION
**Pull request by nicls**
## Thinking Path

> - Paperclip is the open source app people use to manage AI agents for work.
> - Recovery keeps assigned issues aligned with completed agent runs.
> - A successful routine run can finish without a required issue disposition.
> - Paperclip then uses a source-scoped recovery action to request only that disposition.
> - A successful status-only recovery can still leave the routine issue in progress.
> - The stranded-issue classifier then starts a normal productive continuation from that recovery run.
> - This pull request treats that exact lineage as an exhausted handoff.
> - The benefit is that routine recovery stops for an owner decision instead of starting unrelated agent work.

## Linked Issues or Issue Description

- Refs #7496
- Refs #11141
- Refs #9010
- Related open approaches: #6398, #8434, and #12216

## What Changed

- Detect a successful source-scoped missing-disposition recovery for a `routine_execution` issue.
- Route that lineage through the existing exhausted-handoff escalation.
- Keep fresh owner work eligible for normal productive continuation.
- Add integration tests for the blocked recovery path and the fresh-owner boundary.

## Verification

- Reproduced the bug before the fix. The focused test received one productive continuation instead of zero.
- Ran `corepack pnpm --filter @paperclipai/server exec vitest run src/__tests__/heartbeat-process-recovery.test.ts`. All 286 tests passed.
- Ran `corepack pnpm --filter @paperclipai/server exec vitest run src/services/recovery/successful-run-handoff.test.ts`. All 33 tests passed.
- Built `@paperclipai/plugin-sdk` dependencies and ran `server/node_modules/.bin/tsc --noEmit`. The type check passed.
- The standard server type-check wrapper reached the unchanged Rust runner build. It could not continue because this environment has no `cargo` binary.

## Risks

- Risk is low. The classifier requires a routine origin, a successful recovery-action run, and a missing-disposition recovery cause.
- Ordinary issues and fresh owner comments keep the existing continuation behavior.
- This change has no schema, API, or configuration migration.

> For core feature work, check [`ROADMAP.md`](ROADMAP.md) first and discuss it in `#dev` before opening the PR. Feature PRs that overlap with planned core work may need to be redirected — check the roadmap first. See `CONTRIBUTING.md`.

## Model Used

- OpenAI Codex with GPT-5. The exact deployment revision was not available in the session metadata. The model used agentic tool use, repository search, code execution, and test execution.

## Checklist

- [x] I have included a thinking path that traces from project context to this change
- [x] I have specified the model used (with version and capability details)
- [x] I have checked ROADMAP.md and confirmed this PR does not duplicate planned core work
- [x] I have searched GitHub for duplicate or related PRs and linked them above
- [x] I have either (a) linked existing issues with `Fixes: #` / `Closes #` / `Refs #` OR (b) described the issue in-PR following the relevant issue template
- [x] I have not referenced internal/instance-local Paperclip issues or links (only public GitHub `#NNN` / `github.com/paperclipai/paperclip` URLs)
- [x] My branch name describes the change (e.g. `docs/...`, `fix/...`) and contains no internal Paperclip ticket id or instance-derived details
- [x] I have run tests locally and they pass
- [x] I have added or updated tests where applicable
- [x] I have updated relevant documentation to reflect my changes (No documentation change is required.)
- [x] I have considered and documented any risks above
- [ ] All Paperclip CI gates are green
- [ ] Greptile is 5/5 with no open P2s, recommendations, or follow-ups
- [x] I will address all Greptile and reviewer comments before requesting merge

### CODE REVIEW

File: server/src/services/recovery/service.ts [hunk 1]
```diff
@@ -4938,7 +4987,9 @@ export function recoveryService(
         }
         continue;
       }
-      const handoffEvidence = isExhaustedSuccessfulRunHandoff(latestRun);
+      const handoffEvidence =
+        isExhaustedSuccessfulRunHandoff(latestRun) ??
+        routineMissingDispositionRecoveryEvidence(issue, latestRun);
```
[the remark this answers is not in the document]
Comment (nicls) [reply server/src/services/recovery/service.ts]: Valid finding. Fixed in 6466257. Recovery now checks for persisted human direction created after the recovery run before it escalates. When such direction exists, reconciliation uses the productive continuation path. The new test covers the interval where the owner comment is committed but its asynchronous wake is not yet present. The full 286-test recovery file and the server TypeScript check pass.
